data/src/main/java/com/microsoft/azure/kusto/data/exceptions/DataServiceException.java

```diff
@@ -3,26 +3,30 @@
 
 package com.microsoft.azure.kusto.data.exceptions;
 
-public class DataServiceException extends Exception {
+public class DataServiceException extends KustoClientException {
     private String ingestionSource;
-
+    private TriState isPermanent;
     public String getIngestionSource() { return ingestionSource; }
 
     public DataServiceException(String message) {
```

Comment:
Nit: This method is never used in the code. Remove?

Reply:
Its for users to use